cmd/linode-event-sink-slack/main.go

```diff
@@ -36,8 +36,13 @@ func main() {
 
 	api = slack.New(config.Slack.Token)
 
-	// TODO: handle bad channel names
-	channel = getSlackChannelByName(config.Slack.Channel)
+	_channel, err := getSlackChannelByName(config.Slack.Channel)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	channel = _channel
```

Comment:
What is the reasoning behind the `_channel` variable? Is there a benefit over assigning the return value to `channel` from the start?

Reply:
It’s a work around for the compiler throwing a fit about “not using” the `channel` variable, which for some reason only happens when you assign to it and `err` simultaneously. I don’t understand it. ¯\_(ツ)_/¯

Reply:
Turns out it’s a compiler implementation detail about “declaring” a variable within a function body. See: https://stackoverflow.com/questions/68016729/about-unused-variable-in-golang

Reply:
What version of go are you on? I can compile both locally and in the slack-sink docker container. Both are running go 1.18.

https://gist.github.com/stvnjacobs/29cdb062095e709ceca75967c0f22434

Reply:
Dug more into it and this is what I found.
1. It compiles fine on go1.17 (what I _was_ on) and go1.18.
    2. BUT that's only because I later added that logging line, so the function-ally(?) scoped `channel` variable _is_ used then.
    3.  This does _not_ actually set the package-scoped `channel` variable (see below)
4. The reassignment is necessary because `err` is a new variable (previous reference is locally scoped) and thus we have to use `:=` for assignment... which also creates a new `channel` variable.

tl;dr variable scoping always trips me up.

```console
# without using the _channel var
❯ go run ./cmd/linode-event-sink-slack
2022/04/06 09:26:34 INFO {channel=C0REDACTED} channel found
2022/04/06 09:26:40 Failed posting to {channel=}: channel_not_found
``` 

```console
# with using the _channel var
❯ go run ./cmd/linode-event-sink-slack
2022/04/06 09:32:37 INFO {channel=C0REDACTED} channel found
2022/04/06 09:32:41 INFO {channel=C0REDACTED} message successfully sent
```